File: codebase/code/ch08_05/android/app/src/main/java/com/masilotti/hikingjournal/components/ButtonComponent.kt
package com.masilotti.hikingjournal.components

import androidx.compose.material3.Text
import androidx.compose.ui.platform.ComposeView
import dev.hotwire.core.bridge.BridgeComponent
import dev.hotwire.core.bridge.BridgeDelegate
import dev.hotwire.core.bridge.Message
import dev.hotwire.navigation.destinations.HotwireDestination
import dev.hotwire.navigation.fragments.HotwireFragment

class ButtonComponent(
    name: String,
    private val bridgeDelegate: BridgeDelegate<HotwireDestination>
) : BridgeComponent<HotwireDestination>(name, bridgeDelegate) {
    private val fragment: HotwireFragment
        get() = bridgeDelegate.destination.fragment as HotwireFragment

    override fun onReceive(message: Message) {
        val composeView = ComposeView(fragment.requireContext()).apply {
            setContent {
                Text("Sign in")
            }
        }
    }
}

File: codebase/code/ch07_11/ios/App/Components/ButtonComponent.swift
import HotwireNative
import UIKit

class ButtonComponent: BridgeComponent {
    override class var name: String { "button" }

    override func onReceive(message: Message) {
        let button = UIBarButtonItem(title: "Sign in")

        let viewController = delegate?.destination as? UIViewController
        viewController?.navigationItem.rightBarButtonItem = button
    }
}
